pkg/util/utils.go

```diff
@@ -723,3 +726,166 @@ func SocketPath() (string, error) {
+					cliConnections = append(cliConnections, args[1])
+					cliConnections = append(cliConnections, args[0])
+				} else { // either image is in first arg, or syntax errors
+					if RemoteArgLength(args[0], 1) > 0 && RemoteArgLength(args[1], 1) > 0 {
+						//in the future, this function could, instead of rejecting renames, also set a DestImageName field
+						return nil, nil, nil, errors.Wrapf(define.ErrInvalidArg, "cannot specify an image rename")
+					} else if RemoteArgLength(args[0], 1) == 0 && RemoteArgLength(args[1], 1) == 0 {
+						return nil, nil, nil, errors.Wrapf(define.ErrInvalidArg, "no image name specified for source or destination")
+					}
+					source.Image = strings.Split(args[0], "::")[1] // this will get checked/set again once we validate connections
+					cliConnections = append(cliConnections, args[0])
+					cliConnections = append(cliConnections, args[1])
+				}
+			} else {
+				//in the future, this function could, instead of rejecting renames, also set a DestImageName field
+				return nil, nil, nil, errors.Wrapf(define.ErrInvalidArg, "cannot specify an image rename")
+			}
+		case strings.Contains(args[1], "::"):
+			if RemoteArgLength(args[1], 1) > 0 {
+				//in the future, this function could, instead of rejecting renames, also set a DestImageName field
+				return nil, nil, nil, errors.Wrapf(define.ErrInvalidArg, "cannot specify an image rename")
+			}
+			err := ValidateImageName(args[0])
+			if err != nil {
+				return nil, nil, nil, err
+			}
+			source.Image = args[0]
+			dest.Remote = true
+			cliConnections = append(cliConnections, args[1])
+		default:
+			//in the future, this function could, instead of rejecting renames, also set a DestImageName field
+			return nil, nil, nil, errors.Wrapf(define.ErrInvalidArg, "cannot specify an image rename")
+		}
+	}
+	return &source, &dest, cliConnections, nil
+}
+
+// validateImageName makes sure that the image given is valid and no injections are occurring
+// we simply use this for error checking, bot setting the image
+func ValidateImageName(input string) error {
+	// ParseNormalizedNamed transforms a shortname image into its
+	// full name reference so busybox => docker.io/library/busybox
+	// we want to keep our shortnames, so only return an error if
+	// we cannot parse what the user has given us
+	_, err := reference.ParseNormalizedNamed(input)
+	return err
+}
+
+// remoteArgLength is a helper function to simplify the extracting of host argument data
+// returns an int which contains the length of a specified index in a host::image string
+func RemoteArgLength(input string, side int) int {
+	if strings.Contains(input, "::") {
+		return len((strings.Split(input, "::"))[side])
+	}
+	return -1
+}
+
+func CreateSCPCommand(podman string, parentFlags []string, command []string, array bool) *exec.Cmd {
+	var cmd *exec.Cmd
+	if array { // machinectl requires an array, su requires a single string for the podman command
```

Comment:
Could you just pass in a command and if the len(command)=1 then just use the command>  Eliminating the podman string and array bool?

Reply:
oh thats a good idea, thanks I will use that